### PR TITLE
Enable Hermes CI jobs for 0.71

### PR DESCRIPTION
**Pull request by shwanton**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

* Hermes CI jobs were not running after the 0.71 upgrade

## Changelog

[GENERAL] [FIXED] - Enable Hermes CI jobs for 0.71

## Test Plan

**Builds w/ prebuilt Hermes archive**

`HERMES_ENGINE_TARBALL_PATH="/Users/shawndempsey/Downloads/react-native-artifacts-0.71.0-hermes-ios-debug.tar.gz" USE_HERMES=1 RCT_NEW_ARCH_ENABLED=1 bundle exec pod install --verbose`

CleanShot 2023-04-10 at 10 20 16

### CODE REVIEW

File: .ado/apple-pr.yml [hunk 1]
```diff
@@ -108,90 +108,90 @@ jobs:
...
-        #   packager_platform: 'macos'
-        #   xcode_sdk: macosx
-        #   xcode_scheme: 'RNTester-macOS'
-        #   xcode_configuration: 'Release'
-        #   xcode_destination: 'platform=macOS,arch=x86_64'
-        #   xcode_actions: 'build'
-        #   xcode_extraArgs: '-xcconfig $(Build.Repository.LocalPath)/.ado/xcconfig/release_overrides.xcconfig' 
-        #   use_fabric: '1'
-        #   use_hermes: '1'
+        Disable Hermes Jobs for now
```
Comment (Saadnajmi) [inline .ado/apple-pr.yml:114]: Delete this line

**Comment by Saadnajmi** [comment]
This needs the latest changes from main for some of the CI

**Comment by Saadnajmi** [comment]
The remaining failures are legitimate, now.

**Comment by Saadnajmi** [comment]
Mentioned offline, could we split the Hermes stuff into a separate PR? I think the `NEW_ARCH_ENABLED` stuff is ready to go (and should be back ported to 0.71-stable), but the Hermes errors needs a little follow up.

**Comment by shwanton** [comment]
Confirmed that the hermes-engine is not using the correct pre-built on `main` and this is what is causing the build issues.

To get this working in a repo that is not React Native core, we need to pass in a custom `hermes-engine` prebuilt archive.

Prebuilts are created for each release & can be downloaded on the release page. 
example for 0.71.0: https://github.com/facebook/react-native/releases/tag/v0.70.1
archive: https://github.com/facebook/react-native/releases/download/v0.70.1/hermes-runtime-darwin-v0.70.1.tar.gz

Once the prebuilt is downloaded, you can pass it to the build like this:
`HERMES_ENGINE_TARBALL_PATH="/Users/shawndempsey/Downloads/react-native-artifacts-0.71.0-hermes-ios-debug.tar.gz" USE_HERMES=1 bundle exec pod install --verbose`

To get this working for CI, a new step is needed to download the prebuilt archive, assign the path to a variable & pass this to the `pod install` command.

**Comment by shwanton** [comment]
Now that we are on 0.72, we should figure out how to pull down pre-builts instead of trying to build from scratch.
The core team doesn't want more complexity around the Hermes build process for 0.71

**Pull request closed without merging**